### PR TITLE
chore(dial): Port over dial from synchro charts

### PR DESCRIPTION
**Pull request by diehbria**
## Overview
Ports over dial component from synchrocharts. Also refactors to remove calls to calculate bounding boxes and eliminate a lot of state mutations, in preference for a simpler, more straightforward implementation

Follow up PR will be made which will add icons, tests, etc. This is the first iteration, and does not expose the dial component publicly. 
## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/react-components/src/components/dial/dialSvg.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+              <tspan fontSize={Math.min(settings.unitFontSize, valueFontSize)} dx={UNIT_SPACE}>
+                {unit}
+              </tspan>
+            )}
+          </tspan>
+        </text>
+      )}
+      {isLoading && (
+        <text y='55%' x='50%' textAnchor='middle' fontWeight='bold' fontSize={settings.fontSize}>
+          Loading
```
Comment (NorbertNader) [inline packages/react-components/src/components/dial/dialSvg.tsx:63]: shouldn't this be internationalized?

Comment (diehbria) [reply packages/react-components/src/components/dial/dialSvg.tsx:63]: This PR is incremental, but yet loading will be internationalized. I want to pull over the `useMessageOverrides` hook as a part of iot app kit API and rethink how we are handling internationalization from the perspective of component contributors

File: packages/react-components/src/components/dial/useArcs.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+export const useArcs = ({
+  percent,
+  radius,
+  lineThickness,
+}: {
+  percent: number;
+  radius: number;
+  lineThickness: number;
+}) => {
+  const [colorRing, setColorRing] = useState('');
```
Comment (NorbertNader) [inline packages/react-components/src/components/dial/useArcs.ts:35]: inconsistent types, here it is string but on line 49 it is `DefaultArcObject`

File: packages/react-components/src/components/dial/useArcs.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { DefaultArcObject, arc } from 'd3-shape';
```
Comment (NorbertNader) [inline packages/react-components/src/components/dial/useArcs.ts:1]: Can we add tests for this?

File: packages/react-components/src/components/dial/dialSvg.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          x='50%'
+          textAnchor='middle'
+          fontWeight='bold'
+          fontSize={settings.fontSize}
+          fill={ColorConfigurations.SECONDARY_TEXT}
+        >
+          {NO_VALUE_PRESENT}
+        </text>
+      )}
+      {displayLabel && (
```
Comment (NorbertNader) [inline packages/react-components/src/components/dial/dialSvg.tsx:78]: I don't think this is tested?

Comment (diehbria) [reply packages/react-components/src/components/dial/dialSvg.tsx:78]: will add one more unit test in `DialBase`

Comment (diehbria) [reply packages/react-components/src/components/dial/dialSvg.tsx:78]: added missing test on the case within `DialBase` that completes this coverate

**Pull request closed without merging**